number_theory/prime.h

```diff
@@ -149,6 +150,41 @@ class EulerSieve {
   }
 };
 
+// Generates all coprime pairs of integers under |num_limit| (inclusive).
+// Each pair (x, y) in the results should satisfy num_limit >= x >= y >= 0.
+template <typename T>
+std::vector<std::pair<T, T>> coprime_pairs(const T &num_limit) {
+  static_assert(std::numeric_limits<T>::is_integer,
+                "coprime_pairs argument |num_limit| must be an integer.");
+  if (num_limit <= 0)
+    return {};
+  uint64_t num_limit_u64 = numeric_cast<uint64_t>(num_limit);
+  std::vector<std::pair<T, T>> pairs;
+
+  // Append pair (x, y) to |pairs| if it is under the limit.
+  // The caller should make sure x >= y.
```

Comment:
What about swapping them instead of forcing the caller?
```cpp
if (x < y)
  std::swap(x, y);
...
```

Reply:
It is not necessary, because the lambda function is only called inside coprime_pairs with x>=y .